### PR TITLE
chore: Upgrade Jest, and peer dependencies, to v28 [CLI-124]

### PR DESCRIPTION
**Pull request by j-luong**
## Pull Request Submission

Please check the boxes once done.

The pull request must:

- **Reviewer Documentation**
    - [x] follow [CONTRIBUTING](https://github.com/snyk/cli/blob/master/CONTRIBUTING.md) rules
    - [x] be accompanied by a detailed description of the changes
    - [x] contain a risk assessment of the change (Low | Medium | High) with regards to breaking existing functionality. A change e.g. of an underlying language plugin can completely break the functionality for that language, but appearing as only a version change in the dependencies.
    - [ ] highlight breaking API if applicable
    - [ ] contain a link to the automatic tests that cover the updated functionality.
    - [x] contain testing instructions in case that the reviewer wants to manual verify as well, to add to the manual testing done by the author.
    - [ ] link to the link to the PR for the User-facing documentation
- **User facing Documentation**
    - [ ] update any relevant documentation in gitbook by submitting a gitbook PR, and including the PR link here
    - [ ] ensure that the message of the final single commit is descriptive and prefixed with either `feat:` or `fix:` , others might be used in rare occasions as well, if there is no need to document the changes in the release notes. The changes or fixes should be described in detail in the commit message for the changelog & release notes.
- **Testing**
    - [x] Changes, removals and additions to functionality must be covered by acceptance / integration tests or smoke tests - either already existing ones, or new ones, created by the author of the PR.

## Pull Request Review

All pull requests must undergo a thorough review process before being merged. 
The review process of the code PR should include code review, testing, and any necessary feedback or revisions. 
Pull request reviews of functionality developed in other teams only review the given documentation and test reports. 

Manual testing will not be performed by the reviewing team, and is the responsibility of the author of the PR.

For Node projects: It’s important to make sure changes in `package.json` are also affecting `package-lock.json` correctly.

****************************If a dependency is not necessary, don’t add it.**************************** 

When adding a new package as a dependency, make sure that the change is absolutely necessary. We would like to refrain from adding new dependencies when possible.
Documentation PRs in gitbook are reviewed by Snyk's content team. They will also advise on the best phrasing and structuring if needed.

## Pull Request Approval

Once a pull request has been reviewed and all necessary revisions have been made, it is approved for merging into 
the main codebase. The merging of the code PR is performed by the code owners, the merging of the documentation PR 
by our content writers.

## What does this PR do?
This PR updates Jest from `Jest@^27` to `Jest@^28`. Jest peer dependencies were also updated.

## Where should the reviewer start?
No new tests written as we're just upgrading dev dependencies, not changing/adding functionality.

Validate behaviour is unchanged. This PR impact only the legacy CLI, but includes a major TypeScript version change.

### Medium
Major version upgrades impact only dev dependencies

## How should this be manually tested?
Can be manually tested with a local binary.
- `make clean`
- `make build`
- `.../binary-releases/snyk-<OS>-<ARCH> <COMMAND> <OPTS>`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -979,7 +979,7 @@ jobs:
           name: Running acceptance tests
           command: |
             << parameters.pre_test_cmds >>
-            npm run test:acceptance -- --selectProjects snyk
+            npm run test:acceptance -- --selectProjects coreCli
```
Comment (j-luong) [inline .circleci/config.yml:982]: nitpick: can this change be reverted? It looks like it didn't fix anything.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -108,7 +108,7 @@ Place fixtures in `./test/fixtures`. Keep them minimal to reduce maintenance. Us
 You can run acceptance tests with:
 
 ```
-npm run test:acceptance -- --selectProjects snyk
+npm run test:acceptance -- --selectProjects coreCli
```
Comment (j-luong) [inline CONTRIBUTING.md:111]: nitpick: can this be reverted too?

File: jest.config.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const { createJestConfig } = require('./test/createJestConfig');
 
 module.exports = createJestConfig({
-  displayName: 'snyk',
+  displayName: 'coreCli',
```
Comment (j-luong) [inline jest.config.js:4]: nitpick: here too

File: CONTRIBUTING.md (hunk 2 above)
Comment (thisislawatts) [reply CONTRIBUTING.md:111]: It feels like a step in the right direction though? Moving away from everything being called `snyk`

**Review by Avishagp — APPROVED** [review]

**Pull request merged**